### PR TITLE
chore: Refactors hide embed share setting flag.

### LINKED ISSUE #25232: Move APP_EMBED_VIEW_HIDE_SHARE_SETTINGS_VISIBILITY to Flagsmith
Since we are deprecating FF4J, we need to move this flag to flagsmiith.

### PR DESCRIPTION
**Pull request by rahulbarwal**
## Description
As we are deprecating FF4J system in favour of flagsmith. 
This PR renames APP_EMBED_VIEW_HIDE_SHARE_SETTINGS_VISIBILITY to release_embed_hide_share_settings_enabled
#### PR fixes following issue(s)
Fixes #25232 
> if no issue exists, please create an issue and ask the maintainers about this first
>
>
#### Type of change
- Chore (housekeeping or task changes that don't impact user perception)

#### How Has This Been Tested?
- [x] Cypress
>
>
#### Dev activity
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#speedbreakers-) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#areas-of-interest-)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [ ] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

### CODE REVIEW

File: app/server/appsmith-server/src/main/resources/features/init-flags.xml [hunk 1]
```diff
@@ -9,7 +9,7 @@
                 <param name="emailDomains" value="appsmith.com,moolya.com"/>
             </flipstrategy>
         </feature>
-        <feature uid="APP_EMBED_VIEW_HIDE_SHARE_SETTINGS_VISIBILITY" enable="true"
+        <feature uid="release_embed_hide_share_settings_enabled" enable="true"
```
Comment (hetunandu) [inline app/server/appsmith-server/src/main/resources/features/init-flags.xml:12]: You should remove this entry from this file

**Review by hetunandu — CHANGES_REQUESTED** [review]

File: app/server/appsmith-server/src/main/resources/features/init-flags.xml (hunk 1 above)
Comment (rahulbarwal) [reply app/server/appsmith-server/src/main/resources/features/init-flags.xml]: done.

**Review by hetunandu — DISMISSED** [review]

**Comment by rahulbarwal** [comment]
Hey @shwetha-ramesh , 

This just moves flag to flagsmith. It is to be tested similar to this: https://github.com/appsmithorg/appsmith/pull/24561#issuecomment-1594395091

**Comment by shwetha-ramesh** [comment]
@rahulbarwal QA Done

File: app/client/src/ce/entities/FeatureFlag.ts [hunk 2]
```diff
@@ -23,4 +25,11 @@ export const DEFAULT_FEATURE_FLAG_VALUE: FeatureFlags = {
   ask_ai_js: false,
   ask_ai_sql: false,
   release_embed_hide_share_settings_enabled: false,
+  ab_ds_schema_enabled: false,
+  ab_ds_binding_enabled: false,
+};
+
+export const AB_TESTING_EVENT_KEYS = {
```
Comment (hetunandu) [inline app/client/src/ce/entities/FeatureFlag.ts]: What is this used for?

Comment (rahulbarwal) [reply app/client/src/ce/entities/FeatureFlag.ts]: This came in with release merge, PR #25126

**Review by hetunandu — APPROVED** [review]

**Pull request merged**